packages/paste-core/components/badge/src/types.ts

```diff
@@ -0,0 +1,13 @@
+import {ButtonProps} from '@twilio-paste/button';
+
+export type NamedChild = React.ReactElement<Record<string, any>, React.NamedExoticComponent>;
+
+export type BadgeChildren = NamedChild | React.ReactText | (React.ReactText | NamedChild)[];
+
+export type BadgeVariants = 'info' | 'default' | 'warning' | 'error' | 'success' | 'new';
+export type BadgeBaseElements = 'button' | 'span' | 'a';
+export interface BadgeProps extends Omit<ButtonProps, 'variant' | 'as'> {
```

Comment:
When we do the docs lets revisit this interface. 

We can do some nice things with conditional types to make sure there's no cross over between variants.

Right now, I think I can add button props to a badge that isn't a button, and we can fix that with conditionals